sgkit/distance/api.py

```diff
@@ -60,6 +62,8 @@ def pairwise_distance(
 
         Omit to let dask heuristically decide a good default. A default can
         also be set globally with the split_every key in dask.config.
+    device
+        The architecture to run the calculation on, either of cpu or gpu
```

Comment:
What happens if I call ``device="gpu"`` and I don't have a GPU? We should define the behaviour here. Maybe we should raise a custom error class, so that this can be handled programatically? (So, client code could try to run on the GPU, and if that fails, run on CPU)?

Reply:
It would return `CudaSupportError` maybe we can catch that and return a custom sgkit `CUDANotFoundError` maybe?

Reply:
CudaSupportError is fine, that's a comprehensible error. We can add a local error later if it seems needed. Thanks.